Gruntfile.js

```diff
@@ -59,23 +71,31 @@ function getConfig(file, bundleStrategy = SINGLE_BUNDLE, mode = PRODUCTION_MODE)
   let globalObject = `(typeof window !== 'undefined' ? window : typeof globalThis !== 'undefined' ? globalThis : this)`;
 
   let chunkFilename = undefined;
-  let publicPath = 'auto';
   if (bundleStrategy === CODE_SPLITTING) {
     // Font files for dynamic import. See: webpackChunkName in async.ts
-    chunkFilename = 'vexflow-font-[name].js';
+    chunkFilename = '[name].js';
+  }
+
+  // Control the type of source maps that will be produced.
+  // If not specified, production builds will get high quality source maps, and development/debug builds will get nothing.
+  // See: https://webpack.js.org/configuration/devtool/
+  // In version 3.0.9 this was called VEX_GENMAP.
+  const devtool = process.env.VEX_DEVTOOL || (mode === DEVELOPMENT_MODE ? false : 'source-map');
 
-    // See: https://webpack.js.org/guides/public-path/
-    // publicPath needs more testing! :-(
-    // In some tests, this needs to be './' to work, but in others it needs to be 'auto' to work.
-    // publicPath = './';
-    publicPath = 'auto';
+  let plugins = [new webpack.BannerPlugin(BANNER) /* Add a banner at the top of the file. */];
+  if (CHECK_FOR_CIRCULAR_DEPENDENCIES) {
+    plugins.push(
+      new CircularDependencyPlugin({
```

Comment:
This plugin displays circular dependencies, allowing us to easily debug any build issues.

Reply:
@ronyeh are you tracking some of these process changes somewhere? (I want to make sure our docs are updated for things like `VEX_GENMAP -> VEX_DEVTOOL`.

Reply:
I just added it to the CHANGELOG, and also listed the available environment variables at the top of the Gruntfile.js, so that anyone who inspects the Gruntfile will know what's available. I added an environment variable for toggling the circular dependency check as well.

I checked git blame and VEX_GENMAP was introduced by Aaron in June 2020. At first glance, I assumed VEX_GENMAP was a boolean to toggle source maps on / off. However, it is an arbitrary string that is passed directly to the webpack "devtool" configuration: https://webpack.js.org/configuration/devtool/  For example, it could be a string like "eval-source-map". `VEX_DEVTOOL` might be a clearer name since it corresponds directly to the webpack devtool option.